File: src/main/java/org/jukeboxmc/crafting/recipes/SmeltingRecipe.java
package org.jukeboxmc.crafting.recipes;

import lombok.ToString;
import org.jukeboxmc.item.Item;

/**
 * @author LucGamesYT
 * @version 1.0
 */
@ToString
public class SmeltingRecipe {

    private final Item input;
    private final Item output;

    public SmeltingRecipe( Item input, Item output ) {
        this.input = input;
        this.output = output;
    }

    public Item getInput() {
        return this.input;
    }

    public Item getOutput() {
        return this.output;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/recipe/SmeltingRecipe.kt
package org.jukeboxmc.api.recipe

import org.jukeboxmc.api.item.Item

class SmeltingRecipe(
    private val input: Item,
    private val output: Item,
    private val type: Type
) : Recipe {

    fun getInput(): Item {
        return this.input
    }

    fun getOutput(): Item {
        return this.output.clone()
    }

    fun getType(): Type {
        return this.type
    }

    override fun toString(): String {
        return "SmeltingRecipe(input=$input, output=$output, type=$type)"
    }

    enum class Type {
        FURNACE,
        BLAST_FURNACE,
        SMOKER,
        SOUL_CAMPFIRE,
        CAMPFIRE,
    }

}